tests/testerina-integration-test/src/test/resources/project-based-tests/mock-tests/src/Mock/tests/main_test.bal

```diff
@@ -1,91 +1,236 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
 import ballerina/test;
-import SubtractModule;
+import ballerina/io;
 import ballerina/math;
+import Mock2;
+
+(any|error)[] outputs = [];
+int counter = 0;
 
 //
-// TEST CASES
+// MOCK FUNCTION OBJECTS
 //
 
-// Test Function mocking in Same module
-@test:Config {}
-function test_AddFunction() {
-    int answer = 0;
-    answer = intAdd(5, 3);
-    test:assertEquals(answer, 2, "Mocking did not take place");
+@test:Mock {
+    functionName : "intAdd"
 }
+test:MockFunction mock_intAdd = new();
 
-// Test Function mocking in Different module
-// Calling function directly
-@test:Config {}
-function test_SubtractFunction() {
-    int answer = 0;
-    answer = SubtractModule:intSubtract(5, 3);
-    test:assertEquals(answer, 8, "Mocking did not take place");
+@test:Mock {
+    functionName : "stringAdd"
 }
+test:MockFunction mock_stringAdd = new();
 
-// Test Function mocking in Different module
-// Calling function directly
-@test:Config {}
-function test_SubtractFunctionInSameModule() {
-    int answer = 0;
-    answer = intSubtract(5, 3);
-    test:assertEquals(answer, 2, "Mocking did take place, but mocked the wrong function");
+@test:Mock {
+    functionName: "floatAdd"
 }
+test:MockFunction mock_floatAdd = new();
 
-// Test Function mocking for native functions
-@test:Config {}
-function test_MockNativeFunction() {
-    float answer = 0;
-    answer = math:sqrt(5);
+@test:Mock {
+    moduleName : "ballerina/math",
+    functionName : "absInt"
+}
+test:MockFunction mock_absInt = new();
 
-    test:assertEquals(answer, 125.0, "Mocking did not take place");
+@test:Mock {
+    moduleName : "mock-tests/Mock2",
+    functionName : "intAdd2"
 }
+test:MockFunction mock2_intAdd = new();
 
-// Test Function mocking for Object methods
-Person p1 = new ("Jane", "Doe");
+@test:Mock {
+    functionName : "intAdd3"
+}
+test:MockFunction mock_intAdd3 = new();
 
-@test:Config {
+@test:Mock {
+    moduleName: "ballerina/io",
+    functionName: "print"
 }
-function test_MockObjectFunction() {
-    test:assertEquals(p1.getFullName(), "John Doe", "Mocking did not take place");
+test:MockFunction mock_print = new();
+
+//
+//  MOCK FUNCTIONS
+//
+
+public function mockIntAdd1(int x, int y) returns (int) {
+    return x - y;
 }
 
+public function mockIntAdd2(int a, int b) returns (int) {
+    return a * b;
+}
+
+public function mockIntAdd3(int a, int b) returns (float) {
+    return 10.0;
+}
+
+public function mockIntAdd4(int a) returns (int) {
+    return a;
+}
+
+public function mockIntAdd5((any|error)... args) returns (int) {
+    int sum = 0;
+
+    foreach var arg in args {
+        if (arg is int) {
+            sum -= arg;
+        }
+    }
+
+    return  sum;
+}
+
+//public function mockStringAdd(string str1) returns (string) {
```

Comment:
Do we need this commented code. There are commented blocks in a few bal files.

Reply:
Removed all unnecessary commented blocks